pkg/api/v1beta1/types.go

```diff
@@ -1625,3 +1623,109 @@ type ComponentStatusList struct {
 
 	Items []ComponentStatus `json:"items" description:"list of component status objects"`
 }
+
+// SecurityContext holds security configuration that will be applied to a container.  If a security context
+// is set on the container spec then it must comply with any constraints defined in the SecurityConstraints context
+// it is running in.  If a security context is not supplied and the pod is running under a SecurityConstraints context
+// then a default SecurityContext may be applied.
+type SecurityContext struct {
+	// Capabilities are the capabilities to add/drop when running the container
+	// TODO: will need to refactor this from the container spec to here
+	Capabilities *Capabilities `json:"capabilities,omitempty"`
+
+	// Run the container in privileged mode
+	// TODO: will need to refactor this from the container spec to here
+	Privileged bool `json:"privileged,omitempty"`
+
+	// SELinuxOptions are the labels to be applied to the container
+	// and volumes
+	SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
+
+	// RunAsUser is the UID to run the entrypoint of the container process.  Corresponding option is --user or -u
+	RunAsUser int64 `json:"runAsUser,omitempty"`
+}
+
+// SELinuxOptions are the labels to be applied to the container
+type SELinuxOptions struct {
+	// User --security-opt="label:user:USER"
+	User string `json:"user,omitempty"`
+
+	// Role --security-opt="label:role:ROLE"
+	Role string `json:"role,omitempty"`
+
+	// Type --security-opt="label:type:TYPE"
+	Type string `json:"type,omitempty"`
+
+	// Level --security-opt="label:level:LEVEL"
+	Level string `json:"level,omitempty"`
+
+	// Disabled --security-opt="label:disable"
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+// SecurityConstraints provides the constraints that at security context provider will
+// ensure that applied SecurityContext requests follow.  When a setting is provided in the SecurityConstraints
+// that conflicts with an actual request it will be implementation specific whether that request is
```

Comment:
Is it "implementation defined" or is it explicitly controlled by "enforcementPolicy"?